lua/customroles/werewolf.lua

```diff
@@ -0,0 +1,797 @@
+        end
+    end)
+
+    ----------------------------
+    -- ACTIVE WEREWOLF CHECKS --
+    ----------------------------
+
+    local function CheckForActiveWerewolf()
+        if not timer.Exists("TTTWerewolfTimeChange") then return end
+
+        local werewolf = player.GetLivingRole(ROLE_WEREWOLF)
+        if werewolf then return end
+
+        local night_visibility_mode = werewolf_night_visibility_mode:GetInt()
+        if night_visibility_mode == WEREWOLF_NIGHT_SHOW_IF_HAS_WEREWOLF then
+            timer.Remove("TTTWerewolfTimeChange")
+            if WEREWOLF.isNight then
+                WEREWOLF.ChangeTime(true, false, true)
+            end
+        end
+    end
+
+    hook.Add("PlayerDeath", "Werewolf_PlayerDeath", function(victim, infl, attacker)
+        CheckForActiveWerewolf()
+    end)
+
+    hook.Add("PlayerDisconnected", "Werewolf_PlayerDisconnected", function(ply)
+        CheckForActiveWerewolf()
+    end)
+
+    hook.Add("TTTPlayerRoleChanged", "Werewolf_TTTPlayerRoleChanged", function(ply, oldRole, newRole)
+        CheckForActiveWerewolf()
+
+        if WEREWOLF.isNight then
+            if newRole == ROLE_WEREWOLF and oldRole ~= ROLE_WEREWOLF then
+                local drop_weapons = werewolf_drop_weapons:GetBool()
+                if drop_weapons then
+                    for _, wep in pairs(ply:GetWeapons()) do
+                        local class = WEPS.GetClass(wep)
+                        if class ~= "weapon_zm_improvised" and class ~= "weapon_wwf_claws" and wep.AllowDrop then
+                            ply:DropWeapon(wep)
+                        end
+                    end
+                end
+                ply:Give("weapon_wwf_claws")
+                ply:SelectWeapon("weapon_wwf_claws")
+
+                local transform_model = werewolf_transform_model:GetBool()
+                if transform_model then
+                    oldPlayerModels[ply:SteamID64()] = ply:GetModel()
+                    SetMDL(ply, "models/player/stenli/lycan_werewolf.mdl")
+                end
+            elseif oldRole == ROLE_WEREWOLF and newRole ~= ROLE_WEREWOLF then
+                ply:StripWeapon("weapon_wwf_claws")
+                if oldPlayerModels[ply:SteamID64()] then
+                    SetMDL(ply, oldPlayerModels[ply:SteamID64()])
+                end
+            end
+        end
+    end)
+
+    ---------------------
+    -- DISABLE WEAPONS --
+    ---------------------
+
+    hook.Add("PlayerCanPickupWeapon", "Werewolf_PlayerCanPickupWeapon", function(ply, wep)
+        if not IsValid(wep) or not IsValid(ply) then return end
+        if ply:IsSpec() then return false end
+
+        if WEREWOLF.isNight and ply:IsWerewolf() and WEPS.GetClass(wep) ~= "weapon_wwf_claws" then return false end
+    end)
+
+    hook.Add("PlayerSwitchWeapon", "Werewolf_PlayerCanPickupWeapon", function(ply, old, new)
+        if not IsValid(old) or not IsValid(new) or not IsValid(ply) then return end
```

Comment:
Oh, this is how you're preventing the weapon switch. The comment I put in `SWEP:Holster` might be a cleaner way of doing it.